### PR TITLE
Add a script to generate spack's custom scope configurations from LCG releases

### PR DESCRIPTION
**Pull request by mirguest**
It is convenient to use spack's custom scope to override the defaults scope. So based on the existing template in https://github.com/key4hep/k4-spack/tree/master/config, a script is created to generate the complete packages.yaml from LCG releases.
* It will first source the setup script in LCG view.
* It will then generate the `packages.yaml` file with a complete list of packages found in LCG releases. All the packages will be marked as `buildable: false`.
* It will also generate the `compilers.yaml` file.

In order to change the LCG version and arch, just modify the functions in this script:
```bash
function lcg-version() {
    echo LCG_97_FCC_2
}

function lcg-arch() {
    echo x86_64-centos7-gcc8-opt
}
```

### CODE REVIEW

**Comment by vvolkl** [comment]
Hi Tao Lin,

thanks for this -- just to warn you, from my experience it is not advisable to use the lcg releases as external packages in a spack installation. Also, a colleague has been working on writing recipe for all lcg packages, even those that we don't need in key4hep in spack yet. So in the future it will be possible to use the lcg releases as an upstream repository  in spack.

I noticed that spack has recently added a new command `spack external find` https://spack.readthedocs.io/en/latest/build_settings.html#cmd-spack-external-find . Maybe this can also help simplify the script.

**Comment by mirguest** [comment]
Hi @vvolkl ,
Thank you for your suggestion. The `spack external` looks interesting. 
Tao

**Comment by vvolkl** [comment]
Hi Tao Lin,

do you still want to merge this? In case the lcg releases are used as externals, this script is certainly very useful.
Cheers,
Valentin

**Comment by mirguest** [comment]
Hi @vvolkl , 

It would be great if you could help me merge it. Please let me know if there is any problem in this script.
Thank you.

Tao

**Comment by vvolkl** [comment]
Hi Tao Lin,

ok, I had a closer look at the script now, and have two comments:

* The script is not executable right now, maybe you can do a `chmod +x generate_LCG_packages.sh` and commit it again. Git should keep track of the file permissions.
* The package names in lcg unfortunately are not identical with the ones used by spack. Some examples: `cppunit` in spack is turns into `CppUnit` in the `packages.yaml`, and `acts` shows up in `acts_core` . This might need a lot of manual intervention in the script to get right. I remembered that we had something in FCC that tries to do this: https://github.com/HEP-FCC/fcc-spi/blob/master/create_lcg_package_specs.py I'm afraid it is a bit outdated but maybe it can still be helpful for you.

Let me know if you want to adress these points now or I should just merge.
Cheers,
Valentin

**Comment by mirguest** [comment]
Hi @vvolkl ,

Thank you for pointing me to the FCC script. It is useful. 
I have update the script according to your suggestion.

Tao

File: scripts/generate_LCG_packages.sh [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        Garfield++)pkg=garfield++;;
+        Gaudi)     pkg=gaudi;;
+        Geant4)    pkg=geant4;;
+        GSL)       pkg=gsl;;
+        HeapDict)  pkg=heapdict;;
+        HepMC)     pkg=hepmc;;
+        HepPDT)    pkg=heppdt;;
+        herwig++)  pkg=herwig3;;
+        java)      pkg=jdk;;
+        Jinja2)    pkg=jinja2;;
```
Comment (vvolkl) [inline scripts/generate_LCG_packages.sh:154]:
```suggestion
        Jinja2)    pkg=py-jinja2;;
```

**Comment by vvolkl** [comment]
Excellent, looks good to me now. A few packages you take from lcg actually do not exist in spack yet, so we would have to add recipes. I think that concerns `garfield` and some others. But spack doesn't seem to mind if there are some entries in `packages.yaml` it does not recognize, so I'll merge

**Pull request merged**